File: java/data_src_dependent/bst_eq/bst_eq.java

//
// SimpleDriver.java : Driver to illustrate importing USGS earthquake data
// 
import bridges.connect.Bridges;
import bridges.base.BSTElement;
import bridges.data_src_dependent.EarthquakeUSGS;
import bridges.data_src_dependent.Tweet;
import bridges.data_src_dependent.USGSaccount;
import java.util.List;

public class bst_eq {
	public static final int maxElements = 1000; //number of tweets

	public static void main(String[] args) throws Exception{
	
    							// Instantiate a Bridges object */

		Bridges bridges = new Bridges (Integer.parseInt(args[0]), args[1], args[2]);
		bridges.setServer (args[3]);
		bridges.setTitle("Recent Earthquakes(USGIS Data)");

								// Retrieve a list of (maxElements) Tweets */	
		List<EarthquakeUSGS> eqlist = Bridges.getEarthquakeUSGSData(maxElements );
		BST<Double, EarthquakeUSGS>  bst = new BST<Double, EarthquakeUSGS>();
		for ( int i = 0; i < eqlist.size(); i++ ){
			if (eqlist.get(i).getMagnitude() > 3.)
				bst.insert(eqlist.get(i).getMagnitude(), eqlist.get(i));

		}
		// To do: use the list of earthquake objects and insert them into
		// a binary search tree (use the BST API); use the magnitude as
		// search key

		// Set each node's element with the various attributes of the 
		// earthquake; in the visualization the mousing over the element will
		// reveal this label

		
		bridges.setDataStructure(bst.getTreeRoot());
//		bridges.setVisualizeJSON(true);
		//* Read each Tweet returned from Bridges
//		System.out.println("\n\n\n------Printing Earthquakes-------\n");
//		for ( int i = 0; i < eqlist.size(); i++ ){
//			System.out.println(eqlist.get(i));
//		}
		bridges.visualize();
    }
}

File: c++/site_examples/bst_eq/bst_eq.cpp

#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/EarthquakeUSGS.h"
#include "book.h"
#include "BSTElement.h"
#include "BST.h"


using namespace bridges;

int max_quakes = 1000;

int main(int argc, char **argv) {
	string hilite_color = "orange", 
			def_color = "green",
			end_color = "red";
		
	Bridges *bridges = new Bridges(101, argv[1], argv[2]);

								// read the earth quake  data and build the BST
	bridges->setTitle("Recent Earthquakes (USGIS Data)");

	DataSource *ds = new DataSource;
    vector<EarthquakeUSGS> eq_list = ds->getEarthquakeUSGSData(max_quakes);

	BST<float, EarthquakeUSGS> *bst = new BST<float, EarthquakeUSGS> ();

	for (int k = 0; k < max_quakes; k++) {
		EarthquakeUSGS eq = eq_list[k];
		if (eq.getMagnitude() > 3.)
			bst->insert(eq.getMagnitude(), eq);
	}
	bst->setProperties(bst->getRoot());

					// visualize the binary search tree
	bridges->setDataStructure(bst->getRoot());
	bridges->visualize();

	return 0;
}
